### PR TITLE
Realtime OVMM demo with atomic tasks

### PR DESCRIPTION
**Pull request by hello-atharva**
## Description

Provide a brief summary of the changes introduced in this pull request.

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

File: src/stretch/app/ai_pickup.py [hunk 1]
```diff
@@ -125,6 +125,7 @@ def main(
         robot_ip=robot_ip,
         use_remote_computer=(not local),
         parameters=parameters,
+        enable_rerun_server=False,
```
Comment (hello-cpaxton) [inline src/stretch/app/ai_pickup.py:128]: Why?

File: src/stretch/app/ovmm.py [hunk 2]
```diff
@@ -63,7 +66,7 @@
     type=click.Choice(get_llm_choices()),
 )
 @click.option("--explore-iter", default=10, type=int, help="Number of iterations to explore")
-@click.option("--target_object", type=str, default="toy", help="Type of object to pick up and move")
+@click.option("--target_object", type=str, help="Type of object to pick up and move")
```
Comment (hello-cpaxton) [inline src/stretch/app/ovmm.py:69]: What is this field doing now? why no default/

File: src/stretch/app/ovmm.py [hunk 3]
```diff
@@ -102,29 +106,32 @@ def main(
...
         use_remote_computer=(not local),
         parameters=parameters,
+        enable_rerun_server=True,
     )
-    robot.move_to_nav_posture()
-    agent = RobotAgent(robot, parameters, semantic_sensor)
-    agent.update()
+    agent = RobotAgent(robot, parameters, semantic_sensor, enable_realtime_updates=True)
+    agent.start()
+    time.sleep(5)
```
Comment (hello-cpaxton) [inline src/stretch/app/ovmm.py:115]: why this long sleep

File: src/stretch/app/ovmm.py [hunk 4]
```diff
@@ -102,29 +106,32 @@ def main(
...
-    agent.run_exploration(
-        manual_wait=False,
-        explore_iter=explore_iter,
-        task_goal=target_object,  # arbitrary object to collect
-        # as many instances as possible
-        go_home_at_end=True,
-        visualize=False,
-    )
+    # Run exploration
+    if target_object is not None or explore:
```
Comment (hello-cpaxton) [inline src/stretch/app/ovmm.py:124]: why set explore as a flag like this?

File: src/stretch/mapping/voxel/voxel_map.py [hunk 5]
```diff
@@ -272,7 +273,18 @@ def is_valid(
         y1 = y0 + dim
 
         if obstacles is None:
-            obstacles, explored = self.voxel_map.get_2d_map()
+            attempt = 0
+            max_attempts = 10
```
Comment (hello-cpaxton) [inline src/stretch/mapping/voxel/voxel_map.py:277]: These should be parameters

File: src/stretch/mapping/voxel/voxel_map.py [hunk 6]
```diff
@@ -272,7 +273,18 @@ def is_valid(
...
+            while True:
+                try:
+                    obstacles, explored = self.voxel_map.get_2d_map()
+                    break
+                except Exception as e:
```
Comment (hello-cpaxton) [inline src/stretch/mapping/voxel/voxel_map.py:282]: Try-catch tends to be slow; what exceptions happen here?

File: src/stretch/motion/constants.py [hunk 7]
```diff
@@ -161,7 +161,7 @@ class HelloStretchIdx:
         -1.5,  # wrist pitch
         0.0,  # wrist yaw
         0.0,
-        math.radians(-65),
+        math.radians(-50),
```
Comment (hello-cpaxton) [inline src/stretch/motion/constants.py:164]: this needs to be synced across remote and local - what's the justification behind the change?

File: src/stretch/mapping/voxel/voxel_map.py (hunk 6 above)
Comment (hello-atharva) [reply src/stretch/mapping/voxel/voxel_map.py:282]: Sometimes voxel_map is empty and the method call fails

File: src/stretch/motion/constants.py (hunk 7 above)
Comment (hello-atharva) [reply src/stretch/motion/constants.py:164]: This change is no longer needed with my new robot bounds filter

File: src/stretch/mapping/voxel/voxel_map.py (hunk 6 above)
Comment (hello-cpaxton) [reply src/stretch/mapping/voxel/voxel_map.py:282]: Better to change the function to handle the error properly. try-except is very slow in python

File: src/stretch/motion/constants.py (hunk 7 above)
Comment (hello-cpaxton) [reply src/stretch/motion/constants.py]: ok then we should revert it

File: docs/apps.md [hunk 8]
```diff
@@ -249,3 +249,17 @@ This is an experimental app that uses the voxel map to plan a path to a goal. It
 ```bash
 python -m stretch.app.vlm_planning
 ```
+
+### Open-Vocabulary Mobile Manipulation
+
+This is an experimental app that uses a language model to generate commands for the robot given a natural language input. It is not yet fully functional.
```
Comment (hello-cpaxton) [inline docs/apps.md:255]: What about it is not fully functional?

File: docs/apps.md [hunk 9]
```diff
@@ -249,3 +249,17 @@ This is an experimental app that uses the voxel map to plan a path to a goal. It
...
+
+```bash
+python -m stretch.app.ovmm
+```
+
+Options include:
+
+- `--robot_ip` - The IP address of the robot.
+- `--task` - The task to perform. Can be a natural language command for the robot.
+- `--llm` - The language model to use. Can be `openai`, `gemma2b`, or `llama`.
```
Comment (hello-cpaxton) [inline docs/apps.md:265]: Qwen?

File: src/stretch/config/default_planner.yaml [hunk 10]
```diff
@@ -115,6 +115,13 @@ instance_memory:
   # What doe this mean? If you have a view of a bottle on a table, should we remove the table?
   # It will have an effect on performance.
   mask_cropped_instances: False  # Should we remove the background from the instance views?
+  matching:
+    # How much the instance has to match the query to be considered the same
+    # This is a threshold on the cosine similarity between the instance and the query
+    # Set this value by experimting with:
+    #   python -m stretch.app.query --threshold 0.05
+    # You can change the threshold to anything that makes sense.
+    feature_match_threshold: 0.1
```
Comment (hello-cpaxton) [inline src/stretch/config/default_planner.yaml:124]: Why is this added

File: src/stretch/app/ai_pickup.py (hunk 1 above)
Comment (hello-cpaxton) [reply src/stretch/app/ai_pickup.py:139]: I don't think this should be changed, we should revert this

File: src/stretch/agent/operations/pregrasp.py [hunk 11]
```diff
@@ -32,11 +32,10 @@ def can_start(self):
             return False
 
         start = self.robot.get_base_pose()
-        if not self.navigation_space.is_valid(start):
+        if not self.navigation_space.is_valid(start, verbose=True):
```
Comment (hello-cpaxton) [inline src/stretch/agent/operations/pregrasp.py:35]: Why?

File: src/stretch/agent/operations/pregrasp.py [hunk 12]
```diff
@@ -32,11 +32,10 @@ def can_start(self):
...
             self.error(
                 f"{self.name}: [ERROR]: Robot is in an invalid configuration. It is probably too close to geometry, or localization has failed."
             )
-            breakpoint()
```
Comment (hello-cpaxton) [inline src/stretch/agent/operations/pregrasp.py:39]: What happens if there's no break?

File: src/stretch/dynav/mapping_utils/voxel.py [hunk 13]
```diff
@@ -395,6 +399,28 @@ def add(
...
+            # Remove points that are too close to the robot
+            mesh = self.urdf_visualizer.get_combined_robot_mesh()
+
+            # Fetch bounds of the robot
+            bounds = mesh.bounds()
+
+            # Expand bounds by the local radius
+            bounds = bounds + np.array(
+                [
+                    [-0.1, -0.1],
```
Comment (hello-cpaxton) [inline src/stretch/dynav/mapping_utils/voxel.py:412]: This is a very large step - 10cm - right?

**Review by hello-cpaxton — COMMENTED** [review]
Still some questions to resolve

File: docs/apps.md (hunk 8 above)
Comment (hello-atharva) [reply docs/apps.md:255]: I am still not fully confident about grasping. But the behavior is similar to ai_pickup

File: src/stretch/agent/operations/pregrasp.py (hunk 12 above)
Comment (hello-atharva) [reply src/stretch/agent/operations/pregrasp.py:39]: This change was made before I added the new filter. Reverted.

File: src/stretch/dynav/mapping_utils/voxel.py (hunk 13 above)
Comment (hello-atharva) [reply src/stretch/dynav/mapping_utils/voxel.py:412]: Correct. Trying out with a smaller value and testing with ai_pickup.

**Review by hello-cpaxton — APPROVED** [review]
LGTM
